### DISCUSSION TITLE
Hot reload or remove GraphQL Server (or parameters)

### DISCUSSION THREAD

**Discussion by alexander161198** [ChilliCream/hotchocolate#4902] [Q&A]
Hello!

**Now my project is doing the following:**
When the service starts, query and mutation classes are automatically generated based on the database context. These classes are then added to the application assembly. So, I can make GraphQL queries and mutations to existing database entities.

```
public void ConfigureServices(IServiceCollection services)
        {
            services
                .AddHttpClient()
                .AddCustomDataServiceConfigurations(_configuration)
                .AddScoped<IGraphQLRepository, GraphQLRepository>()
                .AddCustomDataServiceDi(_configuration)
                // add GraphQL support
                .AddGraphQLStore(_configuration);
 ;
        }
```

When AddGraphQLStore method is:
```
public static IServiceCollection AddGraphQLStore(this IServiceCollection services, IConfiguration configuration)
        {
            var dbConf = configuration.GetSection("db").Get<DbConnection>();
            var options = DbModelsGeneratorOptions.GetDefaultOptions(dbConf.ConnectionString, dbConf.Schemas);
            var grQl = services.AddGraphQLServer();
            var modelGenerator = new PgDbModelsGenerator(options);
            var modelCompiler = new InteractiveStoreCompiler();
            services.AddInteractiveQuery(grQl, modelCompiler, modelGenerator, options);

            return services;
        }
```

**What I want to:**
When updating the database (for example, adding a new table), don't restart the service, but use the new method of reloading the context and regenerating query and mutation classes.

**My thoughts on how to do this:**
Remove all services associated with the GraphQL from the IServiceCollection and run AddGraphQLStore again. 
Is there a method opposite to AddGraphQLServer method (that delete GaphQL server from DI) or something similar?

Thanks.

**Comment by tobias-tengler** [comment]
I'm not sure how you've built your dynamic schema, but when using `ITypeModule` you have the `TypesChanged` event that you can trigger to reload the schema.

Currently there is no documentation on this, but you can read [this blog post entry](https://chillicream.com/blog/2021/09/27/hot-chocolate-12#dynamic-schemas) as a primer and look at [this example](https://github.com/ChilliCream/hotchocolate-examples/blob/master/misc/TypeModules/DynamicTypes/CustomTypeModule.cs).

**Reply by alexander161198** [reply]
Thank you for answer. But I don't use `ITypeModule`. So, there is no methods like _AddGraphQLServer_ or _ChangeQueryType_?

**Comment by lunindv** [comment]
This very actual problem for me to. I don't undestend how ITypeModule can help me. We tried use ITypeModule but it not resolve our task. On example i has classes:
1. Query - code for all my resolvers
2. Mutation - code for all my mutations
3. MyDbContext - EF db context
4. Models classes from DB
5. Input and Output classes for my mutations
All classes generates dynamicaly and compile as "in memory" assembly.
And the end i'm  call hotchocolate methods (AddQueryType and AddMutation) for my generated classes using reflection. This work fine.
I want after change structure my DB and regeneration my assemble simple reloading "Query" class and "Mutation" class in runtime. How i can do it?
Can you help me do it?
